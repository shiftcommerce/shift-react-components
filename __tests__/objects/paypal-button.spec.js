// Libraries
import React from 'react'

// Object
import PayPalButton from '../../src/objects/paypal-button'

test('renders correctly', () => {
  // Arrange
  global.paypal = {
    Buttons: {
      driver: jest.fn(() => {
        return jest.fn()
      })
    }
  } // set mock window.paypal
  const paypalCreateOrder = jest.fn()
  const paypalOnApprove = jest.fn()
  const onClick = jest.fn()

  // Act
  const wrapper = shallow(
    <PayPalButton 
      paypalCreateOrder={paypalCreateOrder}
      paypalOnApprove={paypalOnApprove}
      onClick={onClick}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(paypal.Buttons.driver).toHaveBeenCalled()
})
