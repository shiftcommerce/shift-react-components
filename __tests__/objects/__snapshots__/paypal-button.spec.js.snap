// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <div
    id="o-paypal-button-container"
  />
  <mockConstructor
    createOrder={[Function]}
    onApprove={[Function]}
    onClick={[Function]}
  />
</Fragment>
`;
